torchaudio/functional/functional.py

```diff
@@ -479,6 +480,127 @@ def _mel_to_hz(mels: Tensor, mel_scale: str = "htk") -> Tensor:
     return freqs
 
 
+def _hz_to_bark(freq: float, bark_scale: str = "traunmuller") -> float:
+    r"""Convert Hz to Barks.
+
+    Args:
+        freqs (float): Frequencies in Hz
+        bark_scale (str, optional): Scale to use: ``traunmuller``, ``schroeder`` or ``wang``. (Default: ``traunmuller``)
+    Returns:
+        barks (float): Frequency in Barks
+    """
+
+    if bark_scale not in ["schroeder", "traunmuller", "wang"]:
+        raise ValueError('bark_scale should be one of "schroeder", "traunmuller" or "wang".')
+
+    if bark_scale == "wang":
+        return 6.0 * math.asinh(freq / 600.0)
+    elif bark_scale == "schroeder":
+        return 7.0 * math.asinh(freq / 650.0)
+    # Traunmuller Bark scale
+    barks = ((26.81 * freq) / (1960.0 + freq)) - 0.53
+    # Bark value correction
+    if barks < 2:
+        barks += 0.15 * (2 - barks)
+    elif barks > 20.1:
+        barks += 0.22 * (barks - 20.1)
+
+    return barks
+
+
+def _bark_to_hz(barks: Tensor, bark_scale: str = "traunmuller") -> Tensor:
+    """Convert bark bin numbers to frequencies.
+
+    Args:
+        barks (Tensor): Bark frequencies
+        bark_scale (str, optional): Scale to use: ``traunmuller``,``schroeder`` or ``wang``. (Default: ``traunmuller``)
+
+    Returns:
+        freqs (Tensor): Barks converted in Hz
+    """
+
+    if bark_scale not in ["schroeder", "traunmuller", "wang"]:
+        raise ValueError('bark_scale should be one of "traunmuller", "schroeder" or "wang".')
+
+    if bark_scale == "wang":
+        return 600.0 * torch.sinh(barks / 6.0)
+    elif bark_scale == "schroeder":
+        return 650.0 * torch.sinh(barks / 7.0)
+    # Bark value correction
+    if any(barks < 2):
+        idx = barks < 2
+        barks[idx] = (barks[idx] - 0.3) / 0.85
+    elif any(barks > 20.1):
+        idx = barks > 20.1
+        barks[idx] = (barks[idx] + 4.422) / 1.22
+
+    # Traunmuller Bark scale
+    freqs = 1960 * ((barks + 0.53) / (26.28 - barks))
+
+    return freqs
+
+
+def barkscale_fbanks(
+    n_freqs: int,
+    f_min: float,
+    f_max: float,
+    n_barks: int,
+    sample_rate: int,
+    bark_scale: str = "traunmuller",
+) -> Tensor:
+    r"""Create a frequency bin conversion matrix.
+
+    .. devices:: CPU
+
+    .. properties:: TorchScript
+
+    Note:
+        For the sake of the numerical compatibility with librosa, not all the coefficients
+        in the resulting filter bank has magnitude of 1.
+
+        .. image:: https://download.pytorch.org/torchaudio/doc-assets/bark_fbanks.png
+           :alt: Visualization of generated filter bank
+
```

Comment:
this note can be removed -- it's not supported in librosa, correct? and we also dont have tests or the asset image

Reply:
You are right, librosa does not support the bark scale. On the other hand, the asset image is easy to get. I modified the Feature Extractor tutorial to show them. If you want to take a look at it, the file can be downloaded from this [link](https://drive.google.com/file/d/12CZNTk7nPt7qgubASWoyh9ae8OZxOrDA/view?usp=share_link).

Reply:
I do not have access to the link, could you share it with the public or send it to me carolinechen@meta.com?

Reply:
You should have access now.

Reply:
cool thanks! I can help add that to our assets and docs in a future PR

Reply:
Great!, thanks.